sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1215,6 +1216,17 @@ private static void validateResource(Resource resource) {
             headers.put(HttpConstants.HttpHeaders.SCRIPT_ENABLE_LOGGING, String.valueOf(true));
         }
 
+        if (options.getDedicatedGatewayRequestOptions() != null &&
+            options.getDedicatedGatewayRequestOptions().getMaxIntegratedCacheStaleness() != null) {
+            Duration maxIntegratedCacheStaleness =
+                options.getDedicatedGatewayRequestOptions().getMaxIntegratedCacheStaleness();
+            if (maxIntegratedCacheStaleness.toNanos() > 0 && maxIntegratedCacheStaleness.toMillis() <= 0) {
```

Comment:
Can it be a negative value in Java and should we guard against that?

Reply:
It can be negative in Java, but we usually don't have any checks around negative values in java SDK, I guess because of the reason that it is highly unlikely that `Duration` is ever used for negative values in Java industry. 
But I don't have any strong opinion on this, so added the negative value check as well.